File: 2ndLifeFront/src/app/services/customer.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { StorageService } from '../storage/storage.service';


const customerUrl = 'http://localhost:8083';

@Injectable({
  providedIn: 'root'
})
export class CustomerService {

  constructor(private httpClient: HttpClient, private storageService: StorageService  // Assurez-vous que StorageService est injecté ici
  ) { }

  getAllCourses(): Observable<any>{
    return this.httpClient.get(customerUrl + "/api/customer/courses");
  }

  getCourseById(courseId:number): Observable<any>{
    return this.httpClient.get(customerUrl + "/api/customer/course/" + courseId );
  }

  enrollCourse(enrollCourseDto: any): Observable<any>{
    console.log(enrollCourseDto);
    
    return this.httpClient.post(customerUrl + "/api/customer/course/enroll" , enrollCourseDto );
  }

  getAllTeachers(): Observable<any>{
    return this.httpClient.get(customerUrl + "/api/customer/teachers");
  }

  getEnrollmentsByUserId(): Observable<any>{
    //const userId = StorageService.getUserId();
    return this.httpClient.get(customerUrl + "/api/customer/course/enrollments/" + StorageService.getUserId());
    console.log(StorageService.getUserId())

  }
  
}

File: eduLine-Backend/src/main/java/com/example/perfectionnement/eduLine/services/customer/CustomerService.java
package com.example.perfectionnement.eduLine.services.customer;

import java.util.List;



import com.example.perfectionnement.eduLine.dto.CourseDto;
import com.example.perfectionnement.eduLine.dto.CourseDtoAfter;
import com.example.perfectionnement.eduLine.dto.EnrollCourse;
import com.example.perfectionnement.eduLine.dto.EnrollCourseDto;
import com.example.perfectionnement.eduLine.dto.TeacherDto;

public interface CustomerService {
	
	List<CourseDto> getAllCourses();

	List<TeacherDto> getAlTeachers();
	
	boolean enrollCourse(EnrollCourse enrollCourseDto);
	
	CourseDto getCourseById(Long courseId);

	List<EnrollCourseDto> getEnrolledCoursesByUserId(Long userId);
}
